plugins/speccy.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+
+from cloudbot import hook
+
+import re
+import requests
+
+from bs4 import BeautifulSoup as BS
+from contextlib import closing
+
+url_re = \
+    re.compile(
+        'https?:\/\/speccy\.piriform\.com\/results\/[a-zA-Z0-9]+', re.I)
+GPU_RE = \
+    re.compile(
+        '.*(amd|radeon|intel|integrated|nvidia|geforce|gtx).*\n.*', re.I)
+PICO_RE = \
+    re.compile(
+        '.*pico', re.I)
+KMS_RE = \
+    re.compile(
+        '.*kms', re.I)
+BOOSTER_RE = \
+    re.compile(
+        '.*booster', re.I)
+REVIVER_RE = \
+    re.compile(
+        '.*reviver', re.I)
+KILLER_RE = \
+    re.compile(
+        '.*killer.+service', re.I)
+
+
+@hook.regex(url_re)
+def get_speccy_url(match):
+
```

Comment:
Extraneous whitespace

Reply:
Can't pinpoint what you see, as again, it's since been the same many revisions ago.

Reply:
I mean the extra line after the `def` line. I just didn't catch it in earlier reviews.